### PR TITLE
ACM-2124 Disable add node pool and manage node pool buttons when upgrading HC 

### PR DESCRIPTION
**Pull request by o-farag**
Signed-off-by: Omar Farag <ofarag@redhat.com>
(cherry picked from commit a78a4940abda5303d2abbb6976bad2d2b63eba9c)

Disables 'Add node pool' button and 'Manage node pool` when hosted cluster is upgrading

Issue: https://issues.redhat.com/browse/ACM-2124

image

image

### CODE REVIEW

File: frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/NodePoolsProgress.tsx [hunk 1]
```diff
@@ -143,11 +143,14 @@ const NodePoolsProgress = ({ nodePools, ...rest }: NodePoolsProgressProps) => {
                                             ? t(
                                                   'Add node pool is only supported for AWS. Use the HyperShift CLI to add additional node pools.'
                                               )
+                                            : cluster?.hypershift?.isUpgrading
+                                            ? t('Cannot add node pools during hosted cluster upgrade')
```
[the remark this answers is not in the document]
Comment (o-farag) [reply frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/NodePoolsProgress.tsx]: fixed. 0 code smells now

**Review by chenz4027 — APPROVED** [review]
/lgtm

**Pull request merged**